src/page/i-frame.page.ts

```diff
@@ -2,9 +2,25 @@ import { $, browser, ElementFinder } from 'protractor';
 
 export class IFramePage {
   private iFrame1: ElementFinder;
+  private currentPageTitle: ElementFinder;
 
   constructor() {
     this.iFrame1 = $('#frame1');
+    this.currentPageTitle = $('.main-header');
+  }
+
+  public async getPageTitle(): Promise<string> {
+    return await this.currentPageTitle.getText();
+  }
+
+  public async switchToFrame(): Promise<void> {
+    this.currentPageTitle = $('#sampleHeading');
```

Comment:
we shouldn't do this in page objects, at least this is something that actually changes within the **same page**

the `pageTitle` within this file should be only for the `IFramePage` the page title of the main page should be on the [main page page object](https://github.com/kliver98/protractor-workshop/blob/master/src/page/personal-information.page.ts)